experimental/dds/tree2/docs/Shared Tree Philosophy.md

```diff
@@ -0,0 +1,194 @@
+# Shared Tree
+
+The goal of the shared tree project is empower developers to create and maintain better collaborative experiences.
+
+To best achieve this goal we plan to deliver a set of libraries that enable developers unfamiliar with the challenges specific to collaborative applications to:
+
+-   Adopt Shared Tree without concern for having to migrate to another solution due to hitting limitations.
+
+    This means that if a developer wants to make a change that interacts with collaboration, the fact that their application is powered by shared tree will not be a limiting factor on what can be achieved.
+    This includes compatibility between different application or version of an applications, scalability, availability, merge quality etc.
+    More concretely, if faced with a new application collaboration requirement, it should be practical to either build it on top of shared tree or extend shared tree to support it.
+    Doing this should not require major reworking of either the application or shared tree implementations, not require extending it in a way that the upstream shared tree would not be willing to maintain.
+
+-   Productively create and maintain collaborative experiences.
+
+    Doing this "productively" means ensuring that a developer new to collaboration will be directed into using patterns that work well for collaborative applications,
```

Comment:
You should consider moving the emphasis on guiding the developer into patterns that work well with collab into the title, as it is very important.

Reply:
Although, given the bullet point below this "Avoid common...", I'm wondering if they should be merged.

Reply:
I'm going to leave this as is for now. Its kind hard to sperate pitfalls avoidances and productivity, but I think they are things we want to address somewhat seperatly (ex: make sure there are good patterns which are productive is different from avoiding bad patterns that produce bad UX, bugs or are simply error prone or hard to maintain)